src/GraphQLParser/ParserContext.cs

```diff
@@ -689,23 +692,48 @@ private GraphQLName ParseName()
 
         private ASTNode? ParseNamedDefinition()
         {
-            return _currentToken.Value switch
-            {
-                "query" => ParseOperationDefinition(),
-                "mutation" => ParseOperationDefinition(),
-                "subscription" => ParseOperationDefinition(),
-                "fragment" => ParseFragmentDefinition(),
-                "schema" => ParseSchemaDefinition(),
-                "scalar" => ParseScalarTypeDefinition(),
-                "type" => ParseObjectTypeDefinition(),
-                "interface" => ParseInterfaceTypeDefinition(),
-                "union" => ParseUnionTypeDefinition(),
-                "enum" => ParseEnumTypeDefinition(),
-                "input" => ParseInputObjectTypeDefinition(),
-                "extend" => ParseTypeExtensionDefinition(),
-                "directive" => ParseDirectiveDefinition(),
-                _ => null
-            };
+            var s = _currentToken.Value.Span;
+
+            if (s.SequenceEqual("query".AsSpan()))
```

Comment:
Would it be faster to convert `"query".AsSpan()` into a constant or anything?  Probably not but just asking.

Reply:
`Span` is a `ref struct` so it can be stored only on stack. Moreover these methods are almost instantaneous.